pkg/scheduler/ugm/manager_test.go

```diff
@@ -702,6 +702,64 @@ func TestDecreaseTrackedResourceForGroupTracker(t *testing.T) {
 	assert.Equal(t, resources.Equals(groupTracker.queueTracker.childQueueTrackers["parent"].resourceUsage, resources.Zero), true)
 }
 
+func TestIncreaseDifferentAppResourceToDiffGroupLinkage(t *testing.T) {
+	// Increase app rsources to different leaf queue, which have different group linkage
+	// Queue setup:
+	// root.parent with group1 limit
+	// root.parent.leaf1 with group2 limit
+	// root.parent.leaf2 with no limit
+
+	// Increase TestApp1 resource to root.parent.leaf1, it should be tracked by group2's groupTracker
+	// Increase TestApp2 resource to root.parent.leaf2, it should be tracked by group1's groupTracker
+
+	setupUGM()
+	user := "user"
+	groups := []string{"group1", "group2"}
+	conf := createConfigWithDifferentGroupsLinkage(groups[0], groups[1], "50", 50, 5)
+	assert.NilError(t, m.UpdateConfig(conf.Queues[0], "root"))
+
+	// Assert inital ugm state
+	assert.Equal(t, len(m.userTrackers), 0, "Initial ugm state shouldn't have no user Tracker created.")
+	assert.Equal(t, len(m.groupTrackers), 2, "Initial ugm state should have 2 groupTracker created.")
+
+	groupTracker1 := m.groupTrackers["group1"]
+	groupTracker2 := m.groupTrackers["group2"]
+
+	// Assert inital GroupTracker state
+	assert.Equal(t, len(groupTracker1.applications), 0, "Initial state shouldn't have any application in GroupTracker.")
+	assert.Equal(t, len(groupTracker2.applications), 0, "Initial state shouldn't have any application in GroupTracker.")
+	assert.Assert(t, groupTracker1.queueTracker.resourceUsage.IsEmpty(), "Initial state shouldn't have any resources tracked in GroupTracker.")
+	assert.Assert(t, groupTracker2.queueTracker.resourceUsage.IsEmpty(), "Initial state shouldn't have any resources tracked in GroupTracker.")
+
+	// Increase resources to different application, different queue
+	userGroup := security.UserGroup{User: user, Groups: groups}
+	usage, err := resources.NewResourceFromConf(map[string]string{"memory": "10", "vcore": "10"})
+	if err != nil {
+		t.Errorf("new resource create returned error or wrong resource: error %t, res %v", err, usage)
+	}
+	m.IncreaseTrackedResource("root.parent.leaf1", TestApp1, usage, userGroup)
+	m.IncreaseTrackedResource("root.parent.leaf2", TestApp2, usage, userGroup)
```

Comment:
Try doing further increase and ensure limit has been limit. Then, try to remove app through decrease case and ensure entry has been removed in corresponding maps.

Reply:
Update in 97d66c8beb6a87d02e6ff3d233012b2f8cafb45b ([Line 766~789](https://github.com/apache/yunikorn-core/commit/97d66c8beb6a87d02e6ff3d233012b2f8cafb45b#diff-31ea862574435965fec182f9616821c52ff5c57a5df0c2e984d9b8ab52a41d21R766-R789))

Reply:
Update to the simple test case.